drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -445,10 +457,14 @@ int stm32_i2c_msg_read(const struct device *dev, struct i2c_msg *msg,
 	stm32_i2c_enable_transfer_interrupts(dev);
 	LL_I2C_EnableIT_RX(i2c);
 
-	k_sem_take(&data->device_sync_sem, K_FOREVER);
+	while (k_sem_take(&data->device_sync_sem,
```

Comment:
Why is this a `while` loop?  The v1 code seems to use `if`.

If it's attempting to force a second `k_sem_take()` to clean up the semaphore value, that's a little subtle and should warrant a comment (and an investigation of why it isn't required for v1).

Reply:
In v1, a reset operation is performed in this situation. In my test, I don't need to reset i2c to make it work again. So, I think it's more elegant to force the call of `stm32_i2c_master_mode_end` which is not called in interrupt. This function give the semaphore `device_sync_sem`, it 's why a `while` loop is used to take it.

Reply:
Elegant, but confusing.  I'd still prefer to see an explanation in the code, or a construct that isn't so confusing, like an if with a second `k_sem_take()` in the body to consume the `give` from the called function.

Also github discarded a top-level comment when I submitted the review from a different view: in the commit message please fix the spelling of `transfer`.